op-service/queue/queue.go

```diff
@@ -0,0 +1,79 @@
+package queue
+
+// Queue implements a FIFO queue.
+type Queue[T any] []T
+
+// Enqueue adds the elements to the back of the queue.
+func (q *Queue[T]) Enqueue(t ...T) {
+	if len(t) == 0 {
+		return
+	}
+	*q = append(*q, t...)
+}
+
+// Dequeue removes a single element from the front of the queue
+// (if there is one) and returns it. Returns a zero value and false
+// if there is no element to dequeue.
+func (q *Queue[T]) Dequeue() (T, bool) {
+	if len(*q) == 0 {
+		var zeroValue T
+		return zeroValue, false
+	}
+	t := (*q)[0]
+	*q = (*q)[1:]
+	return t, true
```

Comment:
```suggestion
	return q.DequeueN(0)
```

Reply:
I'm not sure I want to return a slice from dequeue. I did reuse `DequeueN` here [65882d9](https://github.com/ethereum-optimism/optimism/pull/12180/commits/65882d9149f27e3360aa119ed8df6ed0d402b6c6) but needed a bit more wrangling to return a single element.